controllers/users.js

```diff
@@ -276,18 +277,78 @@ const updateSelf = async (req, res) => {
  * @param res {Object} - Express response object
  */
 const postUserPicture = async (req, res) => {
+  const { file } = req;
+  const { id: userId, discordId } = req.userData;
+  const { coordinates } = req.body;
+  let discordAvatarUrl = "";
+  let imageData;
+  let verificationResult;
+  try {
+    discordAvatarUrl = await generateDiscordProfileImageUrl(discordId);
+  } catch (error) {
+    logger.error(`Error while adding profile picture of user: ${error}`);
+    return res.boom.badImplementation(INTERNAL_SERVER_ERROR);
+  }
   try {
-    const { file } = req;
-    const { id: userId } = req.userData;
-    const { coordinates } = req.body;
     const coordinatesObject = coordinates && JSON.parse(coordinates);
-    const imageData = await imageService.uploadProfilePicture({ file, userId, coordinates: coordinatesObject });
+    imageData = await imageService.uploadProfilePicture({ file, userId, coordinates: coordinatesObject });
+  } catch (error) {
+    logger.error(`Error while adding profile picture of user: ${error}`);
+    return res.boom.badImplementation(INTERNAL_SERVER_ERROR);
+  }
+  try {
+    verificationResult = await userQuery.addForVerification(userId, discordId, imageData.url, discordAvatarUrl);
+  } catch (error) {
+    logger.error(`Error while adding profile picture of user: ${error}`);
+    return res.boom.badImplementation(INTERNAL_SERVER_ERROR);
+  }
+  return res.status(201).json({
+    message: "Profile picture uploaded successfully! " + verificationResult.message,
+    image: imageData,
+  });
+};
+
+/**
+ * Updates the user data
+ *
+ * @param req {Object} - Express request object
+ * @param res {Object} - Express response object
+ */
+
+const verifyUserImage = async (req, res) => {
+  try {
+    const { type: imageType } = req.query;
+    if (!(imageType === "profile" || imageType === "discord")) {
+      throw new Error("Invalid verification type was provided!");
+    }
+    const { id: userId } = req.params;
+    await userQuery.markAsVerified(userId, imageType);
     return res.json({
-      message: "Profile picture uploaded successfully!",
-      image: imageData,
+      message: `${imageType} image was verified successfully!`,
     });
   } catch (error) {
-    logger.error(`Error while adding profile picture of user: ${error}`);
+    logger.error(`Error while verifying image of user: ${error}`);
+    return res.boom.badImplementation(INTERNAL_SERVER_ERROR);
+  }
+};
+
+/**
+ * Updates the user data
+ *
+ * @param req {Object} - Express request object
+ * @param res {Object} - Express response object
+ */
+
+const getUserImageForVerification = async (req, res) => {
+  try {
+    const { id: userId } = req.params;
+    const userImageVerificationData = await userQuery.getUserImageForVerification(userId);
```

Comment:
Instead of using the try-catch block around the current lines, it can be applied specifically around this line.

Reply:
Just one await-ed line thus felt unnecessary.